### PR TITLE
*: use go 1.12 and migrate to go modules

### PR DESCRIPTION
**Pull request by paulfantom**
~Additionally during migration I discovered an incorrect invocation of `uuid.Must` function, which is also fixed in this PR.~

Additionally bumped golangci-lint binary as previous one wasn't aware of go modules.

/cc: @brancz @metalmatze @s-urbaniak @squat

### CODE REVIEW

File: pkg/benchmark/benchmark.go [hunk 1]
```diff
@@ -130,8 +130,9 @@ func New(cfg *Config) (*Benchmark, error) {
 	}
 
 	for i := range b.workers {
+		var err error
 		w := &worker{
-			id:       uuid.Must(uuid.NewV4()).String(),
+			id:       uuid.Must(uuid.NewV4(), err).String(),
```
Comment (s-urbaniak) [inline pkg/benchmark/benchmark.go]: `uuid.Must` receives a `UUID, error` tuple, which `uuid.NewV4()` returns. Here you'll always inject a `nil` valued error.

Does this actually compile at all for you?

Comment (s-urbaniak) [reply pkg/benchmark/benchmark.go]: ... or did the `uuid` library change its upstream signature?

Comment (squat) [reply pkg/benchmark/benchmark.go]: signature hasn't changed: https://godoc.org/github.com/satori/go.uuid#NewV4

Additionally you can see the example on the project's README: https://github.com/satori/go.uuid#example

Comment (squat) [reply pkg/benchmark/benchmark.go]: I think the problem is that you downgraded the library to v1.2.0, which has a different signature: https://github.com/satori/go.uuid/blob/v1.2.0/generator.go#L64

Comment (paulfantom) [reply pkg/benchmark/benchmark.go]: Fixed it by referring to pseudo-version in go.mod file.

**Comment by s-urbaniak** [comment]
small nit: can you split the vendor commit?

**Pull request merged**